### PR TITLE
Shared request barrier

### PR DESCRIPTION
**Pull request by LavrovArtem**

### CODE REVIEW

**Comment by LavrovArtem** [comment]
https://github.com/DevExpress/testcafe-legacy-api/pull/57

File: .eslintrc [hunk 1]
```diff
@@ -42,7 +42,7 @@
     "no-spaced-func": 2,
     "no-trailing-spaces": 2,
     "no-undef-init": 2,
-    "no-unused-expressions": 2,
+    "no-unused-expressions": 0,
```
Comment (miherlosev) [inline .eslintrc:45]: Don't need to turn off this rule for all code. Just add comments where it's necessary.

File: src/browser/provider/built-in/dedicated/chrome/cdp-client/emitters/client-request.ts [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+import { ClientReqEmitter } from '../../../../../../../shared/types';
+import ProtocolProxyApi from 'devtools-protocol/types/protocol-proxy-api';
+import NetworkApi = ProtocolProxyApi.NetworkApi;
+import Protocol from 'devtools-protocol';
+import RequestWillBeSentEvent = Protocol.Network.RequestWillBeSentEvent;
+import LoadingFinishedEvent = Protocol.Network.LoadingFinishedEvent;
+import LoadingFailedEvent = Protocol.Network.LoadingFailedEvent;
+
+
+type ClientReqType = string;
```
Comment (miherlosev) [inline src/browser/provider/built-in/dedicated/chrome/cdp-client/emitters/client-request.ts:11]: The such reductions (`ClientReqType`, `ClientReqEvtListener`) make the code readability worse. 
Please, use the whole words in naming in all places (filenames, variables, types, etc.)

**Review by miherlosev — CHANGES_REQUESTED** [review]
Only naming issues.

File: .eslintrc (hunk 1 above)
Comment (LavrovArtem) [reply .eslintrc:45]: I had to update `@typescript-eslint/parser` to use _nullish coalescing operator (??)_ here:
https://github.com/DevExpress/testcafe/pull/6541/files#diff-09938ba16ab6bb601c1e825c8106cac7f5c1435fbfd39bf231831fbb2ea5355cR26-R28

And after update this rule, it throws an error each time when finding the following code `this.compilerService?.on(...)`.

I'll fix it through the `if` condition.

**Review by miherlosev — APPROVED** [review]

**Review by AndreyBelym — APPROVED** [review]

**Pull request merged**